### PR TITLE
Update remaining issue templates

### PR DESCRIPTION
**Pull request by JeroenWeener**
This PR updates the issue templates, following the rewrite of the bug template. Issue templates now use GitHub Forms and take inspiration from the [Flutter issue templates](https://github.com/flutter/flutter/tree/master/.github/ISSUE_TEMPLATE).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+name: Submit a regression
+description: Report unexpected behavior that worked in a previous version of the plugin.
+title: "[Regression]: "
+labels: ["needs-triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for taking the time to submit a regression!
+        To effectively tackle it, we require as much information as possible. This might seem a lot of information but
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:10]: I would exclude: ` This might seem a lot of information but`

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for taking the time to submit a regression!
+        To effectively tackle it, we require as much information as possible. This might seem a lot of information but 
+        will allow us to quickly get to the root of the regression.
+  - type: checkboxes
+    attributes:
+      label: Is there an existing issue for this?
+      description: Please search to see if an issue already exists for this feature request or proposal.
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:15]: `this feature request or proposal` Aren't all these regressions bugs?

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      options:
+        - label: I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:17]: the -> for?

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+          required: true
+  - type: checkboxes
+    attributes:
+      label: Please select affected platforms
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:21]: platform(s)

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      options:
+      - label: Android
+      - label: iOS
+      - label: Linux
+      - label: macOS
+      - label: Web
+      - label: Windows
+  - type: textarea
+    attributes:
+      label: Old (and correct) behavior
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:31]: Are we asking for old or correct behaviour. These could be two separate things....

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      description: |
+        Briefly but precisely describe what the geolocator plugin was able to do before the regression occurred.
+
+        Consider attaching something showing the old behavior:
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:35]: Consider seems to describe that it is optional, but we want to encourage users to add it? Maybe, please (it helps when...) attach a video/screenshot or codesample showing....

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+         * code samples
+         * images
+         * videos
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Current behavior
+      description: |
+        Briefly but precisely describe what the geolocator plugin is doing now.
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:45]: everything before the but wil be forgotten about.

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        1. ...
+        2. ...
+        3. ...
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Code sample
+      description: |
+        Please create a minimal reproducible sample that shows the problem
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:67]: we use problem/regression throughout, but could we choose one word that describes it best?

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        and attach it below between the lines with the backticks.
+        
+        Note: Please do not upload screenshots of text. Instead, use code blocks
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:70]: note: -> IMPORTANT, it is also not a bad idea to tell the user why this is important.

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Screenshots or Video
+      description: |
+        Upload any screenshots or video of the regression if applicable.
+      value: |
+        <details>
+        <summary>Screenshots / Video demonstration</summary>
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:89]: Screenshots / Video demonstration -> `Screenshots or Video demonstration`

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+        ```
+
+        </details>
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Screenshots or Video
+      description: |
+        Upload any screenshots or video of the regression if applicable.
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:86]: Please consider uploading any relevant screenshots or videos showcasing the regression, if available. These visual aids can greatly assist us in understanding and addressing the issue more effectively.

File: .github/ISSUE_TEMPLATE/a-regression.yml [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+    attributes:
+      label: Current version
+      description: Which version of the plugin are you currently using?
+      placeholder: ex. 1.2.3
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Last version without regression
+      description: In which version of the geolocator plugin was the regression not yet present?
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/a-regression.yml:104]: Could you kindly specify the version of the geolocator plugin in which the regression was not yet present?

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 13]
```diff
@@ -0,0 +1,53 @@
+name: Submit a feature request
+description: Do you have ideas or suggestions for new features? Submit them here!
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:2]: We value your input! If you have any ideas or suggestions for new features, we encourage you to share them here.

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+title: "[Feature Request]: "
+labels: ["needs-triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for taking the time to submit a feature request!
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:9]: you use "take the time" and "taking the time (they already took)".  maybe something like this: Thank you for submitting a feature request! To help us better understand your request, kindly take a moment to provide the following information. Your input is greatly appreciated.

Description of the feature request:
Use case or problem it solves:
Any specific requirements or considerations:
Any additional information or context:
Thank you for your cooperation!

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+        Please take the time to fill out the information below so that we can better understand your request.
+  - type: checkboxes
+    attributes:
+      label: Is there an existing issue for this?
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:13]: issue  -> feature request?

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+      description: Please search to see if an issue already exists for this feature request or proposal.
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:14]: please use consistency like: `feature request` `feature request or proposal` or `proposal`

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+      options:
+        - label: I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
+          required: true
+  - type: checkboxes
+    attributes:
+      label: Please select affected platforms
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:20]: platform(s)

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+    attributes:
+      label: Use case
+      description: |
+        Please tell us the problem you are running into that led to you wanting
+        a new feature.
+
+        Is your feature request related to a problem? Please give a clear and
+        concise description of what the problem is.
+
+        Describe the alternative solutions you've considered. Is there a package
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:38]: Additionally, please share any alternative solutions you have considered. Have you explored existing packages on pub.dev that might already address this problem?

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+      - label: iOS
+      - label: Linux
+      - label: macOS
+      - label: Web
+      - label: Windows
+  - type: textarea
+    attributes:
+      label: Use case
+      description: |
+        Please tell us the problem you are running into that led to you wanting
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:32]: We appreciate your feature request! In order to better understand your needs, please provide details regarding the problem you're encountering that prompted your desire for a new feature.

File: .github/ISSUE_TEMPLATE/d-enhancement-proposal.yml [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+      label: Proposal
+      description: |
+        Please provide a clear and concise description of your idea.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Pitch
+      description: |
+        How will this refactor improve contributors' lives? Input as many details as possible!
```
Comment (TimHoogstrate) [inline .github/ISSUE_TEMPLATE/d-enhancement-proposal.yml:40]: maybe something like this: How will this refactor enhance the lives of contributors? We would greatly appreciate it if you could provide us with as many details as possible. Specifically, please describe the specific benefits, improvements, or advantages that contributors can expect to experience as a result of this refactor. Your comprehensive input will help us better understand the impact and value of implementing this change.

File: .github/ISSUE_TEMPLATE/a-regression.yml (hunk 4 above)
Comment (JeroenWeener) [inline .github/ISSUE_TEMPLATE/a-regression.yml:21]:
```suggestion
      label: Please select affected platform(s)
```

File: .github/ISSUE_TEMPLATE/a-regression.yml (hunk 3 above)
Comment (JeroenWeener) [reply .github/ISSUE_TEMPLATE/a-regression.yml:17]: I believe 'the' is correct here, as people should search 'through' the existing issues, not 'for' them (that's what the link is for)

File: .github/ISSUE_TEMPLATE/a-regression.yml (hunk 5 above)
Comment (JeroenWeener) [reply .github/ISSUE_TEMPLATE/a-regression.yml:31]: I could see this being confusing. I changed it to 'old', to not make any assumptions about correctness.

File: .github/ISSUE_TEMPLATE/a-regression.yml (hunk 7 above)
Comment (JeroenWeener) [reply .github/ISSUE_TEMPLATE/a-regression.yml:45]: I am not sure what you mean. Could you elaborate?

File: .github/ISSUE_TEMPLATE/c-feature-request.yml (hunk 15 above)
Comment (JeroenWeener) [reply .github/ISSUE_TEMPLATE/c-feature-request.yml:13]: I'd prefer 'issue' here, as I don't want people confused about where to look. As not all issues are labeled properly, I'd want to encourage them to look through all issues.

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for submitting a feature request! To help us better understand your request, kindly take a moment to provide the following information. Your input is greatly appreciated.
+
+        Description of the feature request:
+        Use case or problem it solves:
+        Any specific requirements or considerations:
+        Any additional information or context:
```
Comment (mvanbeusekom) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:14]: This makes more sense formatted as a bulleted list:
```suggestion
        - Description of the feature request.
        - Use case or problem it solves.
        - Any specific requirements or considerations.
        - Any additional information or context.
```

File: .github/ISSUE_TEMPLATE/c-feature-request.yml (hunk 15 above)
Comment (mvanbeusekom) [reply .github/ISSUE_TEMPLATE/c-feature-request.yml:19]: Maybe rephrase is so both terms are used:
```suggestion
      label: Is there already an issue requesting this feature?
```

File: .github/ISSUE_TEMPLATE/c-feature-request.yml [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+  - type: textarea
+    attributes:
+      label: Specific requirements or considerations
+      description: |
+        Please provide any specific requirements or considerations that you have for this feature request.
+  - type: textarea
+    attributes:
+      label: Additional information or context
+      description: |
+        Please provide any additional information or context that you have for this feature request.
```
Comment (mvanbeusekom) [inline .github/ISSUE_TEMPLATE/c-feature-request.yml:69]: Include a request for links or resources describing the feature or APIs that people like to see implemented. This saves us a lot of searching.

```suggestion
        Please provide any additional information or context that you have for this feature request. Also links pointing to existing (native) APIs or articles are extremely helpful.
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Looks great Jeroen, just a few small notes.

File: .github/ISSUE_TEMPLATE/c-feature-request.yml (hunk 22 above)
Comment (JeroenWeener) [reply .github/ISSUE_TEMPLATE/c-feature-request.yml:69]: Good point!

**Review by mvanbeusekom — APPROVED** [review]
LGTM, thanks Jeroen.
